includes/compat/subscriptions/class-wc-payment-gateway-wcpay-subscriptions-compat.php

```diff
@@ -120,6 +133,132 @@ public function update_failing_payment_method( $subscription, $renewal_order ) {
 		$this->add_token_to_order( $subscription, $renewal_token );
 	}
 
+	/**
+	 * Include the payment meta data required to process automatic recurring payments so that store managers can
+	 * manually set up automatic recurring payments for a customer via the Edit Subscriptions screen in 2.0+.
+	 *
+	 * @param array           $payment_meta Associative array of meta data required for automatic payments.
+	 * @param WC_Subscription $subscription The subscription order.
+	 * @return array
+	 */
+	public function add_subscription_payment_meta( $payment_meta, $subscription ) {
+		$active_token = $this->get_payment_token( $subscription );
+
+		$payment_meta[ $this->id ] = [
+			self::PAYMENT_METHOD_META_TABLE => [
+				self::PAYMENT_METHOD_META_KEY => [
+					'label' => __( 'Saved payment method', 'woocommerce-payments' ),
+					'value' => empty( $active_token ) ? '' : strval( $active_token->get_id() ),
+				],
+			],
+		];
+
+		// Display select element on newer Subscriptions versions.
+		add_action(
+			sprintf(
+				'woocommerce_subscription_payment_meta_input_%s_%s_%s',
+				WC_Payment_Gateway_WCPay::GATEWAY_ID,
+				self::PAYMENT_METHOD_META_TABLE,
+				self::PAYMENT_METHOD_META_KEY
+			),
+			[ $this, 'render_custom_payment_meta_input' ],
+			10,
+			3
+		);
+
+		return $payment_meta;
+	}
+
+	/**
+	 * Validate the payment meta data required to process automatic recurring payments so that store managers can
+	 * manually set up automatic recurring payments for a customer via the Edit Subscriptions screen in 2.0+.
+	 *
+	 * @param string          $payment_method_id The ID of the payment gateway to validate.
+	 * @param array           $payment_meta      Associative array of meta data required for automatic payments.
+	 * @param WC_Subscription $subscription      The subscription order.
+	 *
+	 * @throws Exception When $payment_meta is not valid.
+	 */
+	public function validate_subscription_payment_meta( $payment_method_id, $payment_meta, $subscription ) {
+		if ( $this->id !== $payment_method_id ) {
```

Comment:
This variable name (`$payment_method_id`) is confusing, since "payment method ID" means a different thing than "payment gateway ID" everywhere else in the code. Do you think it would be a good idea to change it, or let's keep the same name it has in the Subscriptions extension?

Reply:
Yup! Changed it to `payment_gateway_id` in https://github.com/Automattic/woocommerce-payments/pull/892/commits/7d2bb5465d6b20f6ef7e5cef8cb0e7870f72332d.